src/EventStore.Core/Services/Transport/Http/Authentication/ClientCertificateAuthenticationProvider.cs

```diff
@@ -20,11 +21,16 @@ public class ClientCertificateAuthenticationProvider : IHttpAuthenticationProvid
 			var clientCertificate = context.Connection.ClientCertificate;
 			if (clientCertificate is null) return false;
 
+			var remoteIpAddress = context.Connection.RemoteIpAddress;
+			if (remoteIpAddress is null) return false;
+
 			bool hasReservedNodeCN;
 			string clientCertificateCN;
+			
 			try {
+				var clientHostName = Dns.GetHostEntry(remoteIpAddress).HostName;
 				clientCertificateCN = clientCertificate.GetCommonName();
-				hasReservedNodeCN = clientCertificateCN == _certificateReservedNodeCommonName;
+				hasReservedNodeCN = VerifyCertificateReservedNodeCommonName(clientCertificateCN, _certificateReservedNodeCommonName, clientHostName);
```

Comment:
What would be the value of `clientHostName` here?

Reply:
This will provide us the host name of the client machine. On my local machine (3 nodes cluster) I am getting the following value for IP: 127.0.0.1 :- 
"kubernetes.docker.internal"

Reply:
The idea was to use the list of hostnames provided in the gossip seed setting, not the locally resolved hostname.

Here's what I wrote in the issue:
> When node2 connects to node1 could it check if the certificate CN matches with one of the host names in the gossip seed instead of using the reserved certificate common name?

Reply:
@alexeyzimarev if we check the certificate CN against the gossip seed, then it means that another node with a different address could not join the cluster like it can today. I think that this is a pretty rare case, but is that acceptable?